x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -211,6 +214,49 @@ export const signalRulesAlertType = ({
           await ruleStatusService.error(errorString);
           wroteStatus = true;
         }
+        const timestampFieldCaps = await services.scopedClusterClient.fieldCaps({
+          index,
+          fields: timestampFields,
+          allow_no_indices: false,
+          include_unmapped: true,
+        });
+
+        if (
+          !wroteStatus &&
+          params.timestampOverride != null &&
+          (isEmpty(timestampFieldCaps?.body?.fields) ||
+            timestampFieldCaps?.body?.fields[params.timestampOverride] == null ||
+            timestampFieldCaps?.body?.fields[params.timestampOverride]?.unmapped?.indices != null)
+        ) {
+          // if there is a timestamp override and the unmapped array for the timestamp override key is not empty,
+          // partial failure
+          const errorString = `The following indices are missing the timestamp override field "${
+            params.timestampOverride
+          }": ${JSON.stringify(
+            isEmpty(timestampFieldCaps?.body?.fields)
+              ? timestampFieldCaps.body.indices
```

Comment:
```ts
? timestampFieldCaps.body.indices
```

Is this right? You have question marks above it and below and everywhere else but here you don't use question marks with `timestampFieldCaps.body`? So which is right and which is wrong?

I would break out the error messaging into its own functions or try to tuck it away outside of the main controller type logic as it kind of gets in the way of reading the file and it in itself isn't really important to the flow of code.

Also this is a bit misleading as you have variables such as `errorString` but then use `logger.debug`? Is it an error or a debug message?

I would put this into a cleaned up function now or in a follow up PR.

You also might want more custom logic as with all the question marks it looks like you might get log messages with the word `undefined` in them?